homework/shared_ptr/tests/tests.cpp

```diff
@@ -1,192 +1,130 @@
-        {
-            auto result = scooter.get();
-            THEN("empty SharedPtr is returning nullptr through get()")
-            {
-                REQUIRE(result == nullptr);
-            }
-        }
+  GIVEN("Empty SharedPtr") {
+    SharedPtr<int> Number(nullptr);
+    WHEN("Get is returning nullptr") {
+      auto result = Number.get();
+      THEN("empty SharedPtr is returning nullptr through get()") {
+        REQUIRE(result == nullptr);
+      }
     }
+  }
 
-    GIVEN("SharedPtr with int = 100")
-    {
-        int power1{100};
-        int power2{200};
-        SharedPtr<int> SharedPower1(new int(power1));
-        WHEN("resetting scooter with TurboScooter")
-        {
-            int *SharedPower2 = new int(power2);
-            SharedPower1.reset(SharedPower2);
-
-            THEN("ControlBlock Shared Counter is still = 1")
-            {
-                REQUIRE(*(SharedPower1.get()) == power2);
-                REQUIRE(SharedPower1.use_count() == 1);
-            }
-        }
-    }
-    
-    GIVEN("SharedPtr with Scooter(insertValue) where power = 100")
-    {
-        int power{100};
-        SharedPtr<Scooter> scooter(new Scooter(power));
-        bool result;
-        WHEN("Assigning poiter to bool variable")
-        {
-            result = scooter;
-            THEN("When not empty, gives true") { REQUIRE(result); }
-        }
+  GIVEN("SharedPtr with Scooter(insertValue) where power = 100") {
+    int power{100};
+    SharedPtr<int> scooter(new int(power));
+    WHEN("Assigning poiter to bool variable") {
+      bool result = scooter;
+      THEN("When not empty, gives true") { REQUIRE(result); }
     }
-    GIVEN("Empty SharedPtr scooter")
-    {
-        SharedPtr<Scooter> scooter(nullptr);
-        bool result;
-        WHEN("Assigning poiter to bool variable")
-        {
-            result = scooter;
-            THEN("When not empty, gives true") { REQUIRE_FALSE(result); }
-        }
+  }
+  GIVEN("Empty SharedPtr scooter") {
+    SharedPtr<Scooter> scooter(nullptr);
+    WHEN("Assigning a pointer to bool variable") {
+      bool result = scooter;
+      THEN("When not empty, gives true") { REQUIRE_FALSE(result); }
     }
+  }
 
-    GIVEN("Two Shared Ptr, one empty, one not")
-    {
-        int power{100};
-        SharedPtr<Scooter> scooter(new Scooter(power));
-        SharedPtr<Scooter> NewScooter(nullptr);
-        WHEN("Using move assignment operator from SharedPtr")
-        {
-            NewScooter = std::move(scooter);
-            THEN("ControlBlock Shared Counter is incremented to 2")
-            {
-                REQUIRE(NewScooter.get()->power_ == power);
-                REQUIRE(scooter.get() == nullptr);
-            }
-        }
+  GIVEN("Two Shared Ptr, one empty, one not") {
+    int power{100};
+    SharedPtr<int> OldSharedPtr(new int(power));
+    SharedPtr<int> NewSharedPtr(nullptr);
+    REQUIRE(OldSharedPtr.use_count() == 1);
+    REQUIRE(NewSharedPtr.use_count() == 1);
```

Comment:
```suggestion
    REQUIRE(NewSharedPtr.use_count() == 0);
```
AFAIK it should be 0 for nullptr. Please check with std::shared_ptr behavior